Source/FieldSolver/ElectrostaticSolver.cpp

```diff
@@ -158,6 +169,96 @@ WarpX::computePhi (const amrex::Vector<std::unique_ptr<amrex::MultiFab> >& rho,
                    Real const required_precision,
                    int const max_iters) const
 {
+#ifdef WARPX_DIM_RZ
+
+    // Create a new geometry with the z coordinate scaled by gamma
+    amrex::Real const gamma = std::sqrt(1._rt/(1. - beta[2]*beta[2]));
+
+    amrex::Vector<amrex::Geometry> geom_scaled(max_level + 1);
+    for (int lev = 0; lev <= max_level; ++lev) {
+        const amrex::Geometry & geom_lev = Geom(lev);
+        const amrex::Real* current_lo = geom_lev.ProbLo();
+        const amrex::Real* current_hi = geom_lev.ProbHi();
+        amrex::Real scaled_lo[AMREX_SPACEDIM];
+        amrex::Real scaled_hi[AMREX_SPACEDIM];
+        scaled_lo[0] = current_lo[0];
+        scaled_hi[0] = current_hi[0];
+        scaled_lo[1] = current_lo[1]*gamma;
+        scaled_hi[1] = current_hi[1]*gamma;
+        amrex::RealBox rb = RealBox(scaled_lo, scaled_hi);
+        geom_scaled[lev].define(geom_lev.Domain(), &rb);
+    }
+
+    // Setup the sigma = radius
+    // sigma must be cell centered
+    amrex::Vector<std::unique_ptr<amrex::MultiFab> > sigma(max_level+1);
+    for (int lev = 0; lev <= max_level; ++lev) {
+        const amrex::Real * problo = geom_scaled[lev].ProbLo();
+        const amrex::Real * dx = geom_scaled[lev].CellSize();
+        const amrex::Real rmin = problo[0];
+        const amrex::Real dr = dx[0];
+
+        amrex::BoxArray nba = boxArray(lev);
+        nba.enclosedCells(); // Get cell centered array (correct?)
+        sigma[lev] = std::make_unique<MultiFab>(nba, dmap[lev], 1, 0);
+        for ( MFIter mfi(*sigma[lev], TilingIfNotGPU()); mfi.isValid(); ++mfi )
+        {
+            const amrex::Box& tbx = mfi.tilebox();
+            const amrex::Dim3 lo = amrex::lbound(tbx);
+            const int irmin = lo.x;
+            Array4<amrex::Real> const& sigma_arr = sigma[lev]->array(mfi);
+            amrex::ParallelFor( tbx,
+                [=] AMREX_GPU_DEVICE (int i, int j, int /*k*/) {
+                    sigma_arr(i,j,0) = rmin + (i - irmin + 0.5_rt)*dr;
+                }
+            );
+        }
+
+        // Also, multiply rho by radius (rho is node centered)
+        // Note that this multiplication is not undone since rho is
+        // a temporary array.
+        for ( MFIter mfi(*rho[lev], TilingIfNotGPU()); mfi.isValid(); ++mfi )
+        {
+            const amrex::Box& tbx = mfi.tilebox();
+            const amrex::Dim3 lo = amrex::lbound(tbx);
+            const int irmin = lo.x;
+            int const ncomp = rho[lev]->nComp(); // This should be 1!
+            Array4<Real> const& rho_arr = rho[lev]->array(mfi);
+            amrex::ParallelFor(tbx, ncomp,
+            [=] AMREX_GPU_DEVICE (int i, int j, int /*k*/, int icomp)
+            {
+                amrex::Real r = rmin + (i - irmin)*dr;
+                if (r == 0.) {
```

Comment:
Could add a comment here mentioning that the factor `dr/3` is consistent with finite-volume formalism of `amrex`'s Poisson solver?

Reply:
Will do

Reply:
@RemiLehe Here's a question. The way the code is now setup, it first calls `ApplyInverseVolumeScalingToChargeDensity` which divides rho by the radius and then has the loop to multiply rho by the radius. I can skip the divide and multiply and be a bit more efficient, though something that still would need to happen is the wrapping around of the charge deposited below the axis. Does this sound like a reasonable thing to do?

Reply:
Hm, I'll think about it more. For now, I think that this PR can be merged as is. If needed, we could do a second PR to implement the above-mentioned changes.